### DISCUSSION TITLE
lifelib development plan

### DISCUSSION THREAD

**Discussion by fumitoh** [lifelib-dev/lifelib#13] [Ideas]
Here're some to-do items that I want to work on in future lifelib releases:

#### Adding in-force models to `basiclife`

`BasicTerm_S` and `BasicTerm_M` are new business models and their applications are limited to pricing and new business analysis,
so in-force models are desired to enhance the usage of lifelib. 
It won't be hard to extend `BasicTerm_S` to allow in-force model points. `BasicTerm_M` may need some consideration.

#### Adding customization examples in `basiclife`

Add some examples showing how to customize the default models. Show how to put parameters in input files instead of directly inputting them in the models. Show how to use the models with an optimizer in pricing where premiums are determined so that the model points meet certain profitability criteria.

#### Adding variable step length models in `basiclife`

`BasicTerm_S` and `BasicTerm_M` are monthly-step models. Compute-intensive applications may prefer annual-step models.
Variable step-length models, such as monthly in the first few years and annually thereafter, may be an interesting option to consider.

#### Adding a new library for saving product

Add a generic plain saving product that can be used to model variable life/annuity and universal life products.

**Comment by difficult-afternoon** [comment]
Hello,
Really looking forward to the 2 new features: inforce and universal life. Those 2 would be game changer >> allow me to use this for production run.
Please add 1 more feature: " change in reserve" with different set of assumption (with pfad). That would allow me to do Ev and VONB.
Thanks a lot.

**Reply by fumitoh** [reply]
What's pfad?

**Reply by difficult-afternoon** [reply]
margin for adverse deviation = mfad
Basically, reserve is calculated using a more conservative set of assumption (compared to the best estimate assumption which is used to project cashflow). a buffer in case your best estimate is wrong. 
So example: Mfad for expense = 10%, and expense assumption = $10 per year >> to calculate  reserve, you need to generate a cashflow using expense = $11 per year, then discount to get reserve amount.
In normal acturial software, there are multiple set of assumption that the user can input, then 1 would be used to project cashflow, 1 would be used to calculate reserve. There is usually a formula to connect the 2 ( like above 10%) or they could be independent (ie, user input 10 in 1 set, and input 11 in 1 set).

**Comment by fumitoh** [comment]
Pleased to announce the release of lifelib v0.2.0. This release introduces two models, BasicTerm_SE and BasicTerm_ME in the basiclife library. These models are in-force projection models, which project the cashflows of in-force policies at time 0 and new business policies at 0 or at future times.

https://lifelib.io/releases/relnotes_v0.2.0.html
